### PR TITLE
feat: QK norm

### PR DESCRIPTION
**Pull request by HennerM**
QK norm helps to keep attention logits bounded and thus help for stability

### CODE REVIEW

**Review by Ndles — APPROVED** [review]
cool, thanks for that
